### PR TITLE
Adds XD port of Rust/WebAssembly sample

### PR DESCRIPTION
**Pull request by jake-pauls**
**Overview**
This is an implementation of the Rust/WebAssembly sample in XD. Refer to the [original PR](https://github.com/AdobeDocs/uxp-photoshop-plugin-samples/pull/40) for the PS version. The only changes made were the icons, manifest, and overall UI to be more suited to XD. The WebAssembly process and underlying Rust implementation is the same.

### CODE REVIEW

File: wasm-rust-sample/README.md [hunk 1]
```diff
@@ -0,0 +1,76 @@
+# WebAssembly and Rust Example
+
+This plugin is a starting point for leveraging Rust and WebAssembly in your UXP plugins. It comes defined with most of the dependencies you need to get started. As this plugin does rely on the [Rust Programming Language](https://www.rust-lang.org/), an environment configured for Rust development will be required before this will be usable in XD.
+
+**Warning:** This plugin currently crashes XD on ARM devices, including Apple Silicon Macs.
```
Comment (jake-pauls) [inline wasm-rust-sample/README.md:5]: @kerrishotts Decided to leave [this warning](https://github.com/AdobeDocs/uxp-photoshop-plugin-samples/pull/42) in for now, let me know if the XD version runs okay with ARM!

**Review by pklaschka — APPROVED** [review]
LGTM 👍 🎉 !

I have (although I would like to look into it someday 😉 ) never worked with Rust, meaning I'm not qualified to do any actual code review on the Rust side of things.

I think this looks great from the JS side of things and the documentation in the `README.md` (where I feel more comfortable giving an actual review).

I was able to build and run the plugin successfully => it worked on my machine 😜 . I will note, however, that `cargo install wasm-pack` failed on my machine. Instead, I then downloaded an installer from https://rustwasm.github.io/wasm-pack/installer/ and ran `yarn test`, which seemed to work. But I assume that this was some quirk on my local machine => no reason for a negative review 🙂 

The only thing that I will mention is that while it seems to work (I was able to run the plugin, after all), there is no mention of `manifest.json` v4 support in the docs yet. Using an "unofficially supported" version, of course, could be a concern when including something in the official samples repo, but since v4 support in XD is probably "inevitable" anyways (in a good way). I think this shouldn't be a show-stopper => **Approved** from my end 😉 . @amandahuarng @kerrishotts What do you think?

@jake-pauls Thank you for your tremendous work, and I wouldn't even have thought that WebAssembly could be a thing in XD 😮

**Comment by amandahuarng** [comment]
The project compiled and similar to what Pablo said the readme and js files look awesome. I am running an Apple M1 which is probably why every time I load the plugin in UXP developer tools my XD crashes, even when I target the build to Rosetta. I think that warning should be kept in the readme, and other than that it looks great to me!

**Comment by kerrishotts** [comment]
I'm fine w/ v4 manifest here -- XD will be supporting the new manifests here soon enough, and having nearly the same manifest for Xd and Ps tells the cross app story.

**Pull request merged**